### PR TITLE
[firewall] disable firewall by default

### PR DESCRIPTION
**Pull request by superwhd**
The firewall feature should not be enabled on some targets, e.g. ot-daemon. Although leaving it enabled does not break the behavior, it creates confusing logs. https://github.com/openthread/ot-docs/issues/67

Depends-On: openthread/ot-br-posix#1188

### CODE REVIEW

File: src/posix/platform/openthread-posix-config.h [hunk 1]
```diff
@@ -203,7 +203,8 @@
  *
  */
 #ifdef __linux__
-#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE & OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
+#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE & OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE && \
+    !OPENTHREAD_POSIX_CONFIG_DAEMON_ENABLE
```
Comment (simonlingoogle) [inline src/posix/platform/openthread-posix-config.h:207]: I think we also have `OPENTHREAD_POSIX_CONFIG_DAEMON_ENABLE=1` for otbr-agent.

File: src/posix/platform/openthread-posix-config.h [hunk 2]
```diff
@@ -203,7 +203,8 @@
  *
  */
 #ifdef __linux__
-#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE & OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
+#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE && OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE && \
+    !OPENTHREAD_POSIX_CONFIG_DAEMON_ENABLE
```
Comment (wgtdkp) [inline src/posix/platform/openthread-posix-config.h:207]: Shouldn't we use dedicated cmake options (e.g. `OT_DAEMON`) for the ingress filter feature?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

I am fine with disabling firewall by default since it's an advanced feature and may cause confusion and unexpected issues.

**Review by wgtdkp — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**